### PR TITLE
Add support for environment variables for server socket and server url

### PR DESCRIPTION
**Pull request by inf17101**
Issues: #75 

# Definition of Done

The PR shall be merged **only if all of following items were checked.** In case an item is not applicable as described, please provide a short explanation right after the item in question. Separate the explanation by a semicolon and write it as bold text like **; requirements are handled in issue #xyz**:

- [ ] documentation of all modules `/*/doc/swdesign` is up-to-date
- [ ] requirements are up-to-date and requirements for new features have been added and mapped to source and test
- [ ] conform to [unit verification strategy](https://eclipse-ankaios.github.io/ankaios/main/development/unit-verification/)

### CODE REVIEW

**Comment by inf17101** [comment]
PR is not ready yet. I will do a refactoring of the nested conditionals, and add unit tests.

File: agent/doc/swdesign/README.md [hunk 1]
```diff
@@ -80,17 +80,48 @@ The following diagram shows the startup sequence of the Ankaios Agent:
...
 Needs:
 - impl
 - itest
 
+#### Agent supports environment variable for the server url
+`swdd~agent-shall-support-environment-variable-for-server-url~1`
+
+Status: approved
+
+When the environment variable with key `ANKAIOS_SERVER_URL` is set and the user has not provided the server url via cli argument,
```
Comment (windsource) [inline agent/doc/swdesign/README.md:99]: I would prefer, if we cloud derive from the name of an env var the tools it belongs to, e.g.

- `ANK_...` modifies the behavior of the Ankaios cli
- `ANKAGENT_...` modifies the behavior of the Ankaios agent
- `ANKSERVER_...` modifies the behavior of the Ankaios server

So using that notation the variable should be called `ANKAGENT_SERVER_URL`.

Comment (windsource) [reply agent/doc/swdesign/README.md:99]: Also I am not sure if we need a separate requirement for every environment variable that we introduce. Just an idea ...

File: common/src/lib.rs [hunk 2]
```diff
@@ -15,7 +15,10 @@
 pub const CHANNEL_CAPACITY: usize = 20;
 pub const DEFAULT_SOCKET_ADDRESS: &str = "127.0.0.1:25551";
 pub const DEFAULT_SERVER_ADDRESS: &str = "http://127.0.0.1:25551";
+pub const SERVER_URL_ENV_KEY: &str = "ANKAIOS_SERVER_URL";
```
Comment (windsource) [inline common/src/lib.rs:18]: I would move the keys to the respective files `ank/src/cli.rs`.

**Comment by windsource** [comment]
As discussed the environment variables for the cli are high prio but the ones for agent and server are low prio as the cli is called much more often and server and agent are usually only started at system startup.

File: agent/doc/swdesign/README.md (hunk 1 above)
Comment (windsource) [reply agent/doc/swdesign/README.md:99]: @krucod3 do you agree with having prefixes per component?

**Comment by inf17101** [comment]
I close this PR I had problems with reverting the stuff. I open a new PR. The decisions and discussions are mainly covered in the corresponding issue.

**Pull request closed without merging**